sqlglot/optimizer/scope.py

```diff
@@ -399,7 +399,7 @@ def _traverse_scope(scope):
         yield from _traverse_select(scope)
     elif isinstance(scope.expression, exp.Union):
         yield from _traverse_union(scope)
-    elif isinstance(scope.expression, (exp.Lateral, exp.Unnest)):
+    elif isinstance(scope.expression, (exp.Lateral, exp.Unnest, exp.Values)):
```

Comment:
(exp.Unnest, exp.Lateral, exp.Values) is used in many places

can we make the inheritance tree something

class DerivedTable:

class LiteralDerivedTable(DerivedTable): ( or some better name )

and then just check for instance of that in like the 4 places where this is used

Reply:
I came up with `SelectlessDerivedTable`. Still not great but does better communicate the aspect that makes these unique which is that they don't have a `SELECT` expression in their definition.

Reply:
Ended up changing to UDTF (User Defined Table Function).